nuxeo/nuxeo.py

```diff
@@ -504,178 +549,48 @@ def workflows(self):
-        server to compute access statistics for various client types (e.g.
-        browser vs devices).
-        """
-
-        headers = {
-            'Cache-Control': 'no-cache',
-        }
-        if self._auth is not None:
-            headers.update([self._auth])
-        headers.update(self._headers)
-        return headers
-
     def _get_cookies(self):
-        return list(self.cookie_jar) if self.cookie_jar is not None else []
+        return list(self._session.cookies) or []
 
     def _log_details(self, e):
-        if hasattr(e, 'fp'):
-            detail = e.fp.read().decode('utf-8')
-            try:
-                exc = json.loads(detail)
-                message = exc.get('message')
-                stack = exc.get('stack')
-                error = exc.get('error')
-                if message:
-                    self.debug('Remote exception message: %s', message)
-                if stack:
-                    self.debug('Remote exception stack: %r',
-                               exc['stack'], exc_info=True)
-                else:
-                    self.debug('Remote exception details: %r', detail)
-                return exc.get('status'), exc.get('code'), message, error
-            except ValueError:
-                # Error message should always be a JSON message,
-                # but sometimes it's not
-                if '<html>' in detail:
-                    message = e
-                else:
-                    message = detail
-                self.error(message)
-                if isinstance(e, urllib2.HTTPError):
-                    return e.code, None, message, None
-        return None
-
-    def _read_response(self, response, url):
-        info = response.info()
-        s = response.read()
-        content_type = info.get('content-type', '')
-        cookies = self._get_cookies()
-        if content_type.startswith('application/json'):
-            self.trace('Response for %r with cookies %r: %r',
-                       url, cookies, s)
-            return json.loads(s) if s else None
-        else:
-            self.trace('Response for %r with cookies %r has content-type %r',
-                       url, cookies, content_type)
-            return s
+        try:
+            exc = e.response.json()
+            message = exc.get('message')
+            stack = exc.get('stack')
+            error = exc.get('error')
+            if message:
+                logger.exception('Remote exception message: {!s}'.format(message))
+            if stack:
+                logger.exception('Remote exception stack: {!s}'.format(stack))
+            else:
+                logger.exception('Remote exception details: {!s}'.format(exc))
+            return exc.get('status'), exc.get('code'), message, error
+        except ValueError:
+            # Error message should always be a JSON message,
+            # but sometimes it's not
+            logger.error(e)
```

Comment:
Also, never pass the exception itself to the logger. Here you could do:

    logger.error('Response is no JSON', exc_info=True)

Or:

    logger.exception('Response is no JSON')

Because `log.exception` is just an alias of `log.error(msg, exc_info=True)`